### PR TITLE
[filter/pytorch] Add support for tuple type output  @open sesame: 12/27 22:40

### PR DESCRIPTION
**Pull request by anyj0527**
- Support torch script model whose output value is `Tuple`
- Add a simple SSAT test for pytorch model whose input and output are multiple tensors
- Tested with PyTorch v1.6.0-rc1 and v1.10.1

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: ext/nnstreamer/tensor_filter/tensor_filter_pytorch.cc [hunk 1]
```diff
@@ -489,7 +489,8 @@ TorchCore::invoke (const GstTensorMemory *input, GstTensorMemory *output)
     g_assert (outputTensorMeta.num_tensors == output_list.size ());
     int idx = 0;
     for (auto &ivalue_element : output_list) {
-      if (processIValue (ivalue_element, &output[idx++])) {
+      ++idx;
```
Comment (jaeyun-jung) [inline ext/nnstreamer/tensor_filter/tensor_filter_pytorch.cc:492]: Q: index started from 1? plz check this line.

Comment (anyj0527) [reply ext/nnstreamer/tensor_filter/tensor_filter_pytorch.cc]: My mistake. I moved the line, thanks.

File: ext/nnstreamer/tensor_filter/tensor_filter_pytorch.cc [hunk 2]
```diff
@@ -295,59 +295,50 @@ TorchCore::getTensorTypeToTorch (tensor_type tensorType, torch::Dtype *torchType
...
+  c10::IntArrayRef sliced_output_sizes = sliced_output.sizes ();
 
-  if (tensor_shape[0] != 0 && outputTensorMeta.num_tensors != tensor_shape[0]) {
-    ml_loge ("Invalid output meta: different size");
-    return -1;
-  }
-
+  /** when output is a scalar */
   if (tensor_shape[0] == 0) {
     tensor_type otype = getTensorTypeFromTorch (output.scalar_type ());
```
Comment (jaeyun-jung) [inline ext/nnstreamer/tensor_filter/tensor_filter_pytorch.cc:315]: duplicated declaration 
`otype = getTensorTypeFromTorch (output.scalar_type ());`

File: ext/nnstreamer/tensor_filter/tensor_filter_pytorch.cc [hunk 3]
```diff
@@ -489,10 +489,11 @@ TorchCore::invoke (const GstTensorMemory *input, GstTensorMemory *output)
     g_assert (outputTensorMeta.num_tensors == output_list.size ());
     int idx = 0;
```
Comment (jaeyun-jung) [inline ext/nnstreamer/tensor_filter/tensor_filter_pytorch.cc:490]: change type to unsigned int

File: ext/nnstreamer/tensor_filter/tensor_filter_pytorch.cc (hunk 2 above)
Comment (anyj0527) [reply ext/nnstreamer/tensor_filter/tensor_filter_pytorch.cc]: Thanks! I fixed the code.

File: ext/nnstreamer/tensor_filter/tensor_filter_pytorch.cc [hunk 4]
```diff
@@ -378,12 +369,13 @@ TorchCore::getOutputTensorDim (GstTensorsInfo *info)
 /**
  * @brief	process the IValue after forward and extract data from ivalue.
  * @param[in] value IValue containing the output in tensor form
- * @param[out]  output Output tensor memory
+ * @param[out] output Output tensor memory
+ * @param[out] idx index of output
```
Comment (dongju-chae) [inline ext/nnstreamer/tensor_filter/tensor_filter_pytorch.cc:373]: `idx` seems to be one of input parameters.

Comment (anyj0527) [reply ext/nnstreamer/tensor_filter/tensor_filter_pytorch.cc]: Thanks! I fixed 'out' with 'in'

**Review by jaeyun-jung — APPROVED** [review]

**Review by myungjoo — APPROVED** [review]

**Pull request merged**